bittensor/core/subtensor.py

```diff
@@ -685,16 +685,8 @@ def determine_chain_endpoint_and_network(
 
         if network is None:
             return None, None
-        if network in ["finney", "local", "test", "archive"]:
-            if network == "finney":
-                # Kiru Finney staging network.
-                return network, settings.FINNEY_ENTRYPOINT
-            elif network == "local":
-                return network, settings.LOCAL_ENTRYPOINT
-            elif network == "test":
-                return network, settings.FINNEY_TEST_ENTRYPOINT
-            elif network == "archive":
-                return network, settings.ARCHIVE_ENTRYPOINT
+        if network in settings.NETWORKS:
+            return network, settings.NETWORK_MAP[network]
```

Comment:
```suggestion
        if chain_endpoint := settings.NETWORK_MAP.get(network):
            return network, chain_endpoint
```

Reply:
we check the `network` argument before `chain_endpoint` if defined